src/utils/code_etl/repo_to_json.py

```diff
@@ -0,0 +1,252 @@
+        os.chdir(self.cwd)
+
+
+def process_local_repo(location, output_dir, repo_name):
+    """Convert a local repository to a series of JSON objects.
+
+    Args:
+        location (str): The path to a local repository.
+        output_dir (str): The path to the directory to save the output files
+            to.
+        repo_name (str): The name to save to the JSON objects as the repository
+            name.
+
+    Returns:
+        None
+    """
+    with cd(location):
+        output_file = output_dir + "/" + repo_name.replace('/', '_') + ".json"
+        all_lines = []
+        with open(output_file, 'w') as f:
+            for file in get_filelist(location):
+                for line in btj.file_to_json(file, repo_name):
+                    f.write(line + "\n")
+
+
+def get_local_repo_name(location):
+    """The basename of a local repository.
+
+    If a local repository is located at:
+
+        /path/to/local/repo/
+
+    This function will return "repo".
+
+    This function must be called from within the repository, so using cd() to
+    change the directory is advised.
+
+    Args:
+        location (str): The path to a local repository.
+
+    Returns:
+        str: The basename of the repository.
+
+    """
+    with cd(location):
+        command = [
+            "git",
+            "rev-parse",
+            "--show-toplevel",
+        ]
+        repo_name = subprocess.check_output(command)
+        base = os.path.basename(repo_name).strip()
+        return base
+
+
+def is_github_like(repo):
+    """Determine if a string looks like a github repository location.
+
+    A github repository takes the form of "word/word". This function tests that
+    there is exactly one '/', and that there are characters on either side. If
+    this is the case, it returns True, otherwise false.
+
+    Args:
+        repo (str): A string indicating the repository location.
+
+    Returns:
+        bool: True if the string looks like a github repository location, False
+        otherwise.
+
+    """
+    split_repo = repo.split('/')
+    if len(split_repo) != 2:
+        return False
+    if not len(split_repo[0]) or not len(split_repo[1]):
+        return False
+
+    return True
+
+
+if __name__ == "__main__":
+
+    import argparse
+    # Set up command line flag handling
+    parser = argparse.ArgumentParser(
+        description="Download and parse a git repository",
+    )
+    parser.add_argument(
+        'repo_location',
+        type=str,
+        help="the location of the repository, either as a github name or a\
+        local file path",
+    )
+    parser.add_argument(
+        'output_directory',
+        type=str,
+        action="store",
+        help="the directory to save the output JSON files",
+    )
+    parser.add_argument(
+        '-n',
+        '--repo-name',
+        type=str,
+        action="store",
+        help="override the default repository name to save to the JSON file"
+    )
```

Comment:
Why do we want to change the repo's name? In what scenario, are we going to use that functionality? Is this for testing in the case when we don't have enough dataset, we will rename a copy with another name to determine whether or not the two repos are alike for better recommendation?